From 146893808ed5e7fed46d597d0ba795ee22a98bcb Mon Sep 17 00:00:00 2001
From: RickyWu-wiwynn <ricky_cx_wu@wiwynn.com>
Date: Fri, 1 Nov 2024 13:32:56 +0800
Subject: [PATCH 1005/1016] arm: dts: aspeed: yosemite4: set TACH config for
 MAX31790 on Fan Boards

Add the setting for max31790 to config PWM as TACH.
- Haven't contributed because it is depend on the following patch:
https://github.com/facebookexternal/openbmc.wiwynn/blob/main/common/recipes-kernel/linux/6.6/0301-hwmon-max31790-support-to-config-PWM-as-TACH.patch
---
 .../aspeed/aspeed-bmc-facebook-yosemite4.dts  | 36 +++++++++++++++++++
 1 file changed, 36 insertions(+)

diff --git a/arch/arm/boot/dts/aspeed/aspeed-bmc-facebook-yosemite4.dts b/arch/arm/boot/dts/aspeed/aspeed-bmc-facebook-yosemite4.dts
index d216505fb2c4..9f67113d9368 100644
--- a/arch/arm/boot/dts/aspeed/aspeed-bmc-facebook-yosemite4.dts
+++ b/arch/arm/boot/dts/aspeed/aspeed-bmc-facebook-yosemite4.dts
@@ -1251,6 +1251,15 @@ adc@1f {
 			pwm@20 {
 				compatible = "maxim,max31790";
 				reg = <0x20>;
+				channel@4 {
+					reg = <4>;
+					sensor-type = "TACH";
+				};
+
+				channel@5 {
+					reg = <5>;
+					sensor-type = "TACH";
+				};
 			};
 
 			gpio@22 {
@@ -1263,6 +1272,15 @@ gpio@22 {
 			pwm@2f {
 				compatible = "maxim,max31790";
 				reg = <0x2f>;
+				channel@4 {
+					reg = <4>;
+					sensor-type = "TACH";
+				};
+
+				channel@5 {
+					reg = <5>;
+					sensor-type = "TACH";
+				};
 			};
 
 			adc@33 {
@@ -1299,6 +1317,15 @@ adc@1f {
 			pwm@20 {
 				compatible = "maxim,max31790";
 				reg = <0x20>;
+				channel@4 {
+					reg = <4>;
+					sensor-type = "TACH";
+				};
+
+				channel@5 {
+					reg = <5>;
+					sensor-type = "TACH";
+				};
 			};
 
 			gpio@22 {
@@ -1311,6 +1338,15 @@ gpio@22 {
 			pwm@2f {
 				compatible = "maxim,max31790";
 				reg = <0x2f>;
+				channel@4 {
+					reg = <4>;
+					sensor-type = "TACH";
+				};
+
+				channel@5 {
+					reg = <5>;
+					sensor-type = "TACH";
+				};
 			};
 
 			adc@33 {
-- 
2.47.3

